sympy/functions/elementary/tests/test_integers.py

```diff
@@ -126,6 +126,14 @@ def test_floor():
     assert Eq(floor(y), y - frac(y))
     assert Eq(floor(y), -ceiling(-y))
 
+    neg = Symbol('n', negative=True)
+    pos = Symbol('p', positive=True)
+
+    assert (floor(neg) < 0) == True
+    assert (floor(neg) >= 0) == False
+    assert (floor(pos) >= 0) == True
+    assert (floor(pos) < 0) == False
+
```

Comment:
```suggestion

    nn = Symbol('nn', nonnegative=True)
    np = Symbol('np', nonpositive=True)
    assert (floor(nn) < 0) == False
    assert (floor(nn) >= 0) == True
    assert (floor(np) <= 0) == True
    assert (floor(np) > 0) == False
```

Reply:
And we should be able to assert than `floor(neg) <= -1` and `ceiling(pos) >= 1`.

Reply:
Right! One can probably look at the code for `frac` to see how I did it there (seems like comparing to one is crucial...). I think that requires to work with the comparisons functions though, so maybe a different PR? (Of course OK to add it here...)

Reply:
I actually approached this whole issue by using the comparison operators...but I think this approach is good here. And the comparison with 1 can be part of a different PR.